balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -1249,6 +1249,46 @@ func (s) TestInitialIdle(t *testing.T) {
 	}
 }
 
+// TestSubBalancerStateLazyUpdate covers the case that if the child reports a
+// transition from TF to Connection, the overall state will still be TF.
+func (s) TestSubBalancerStateLazyUpdate(t *testing.T) {
+	cc := &tcc{TestClientConn: testutils.NewTestClientConn(t)}
+
+	wtb := wtbBuilder.Build(cc, balancer.BuildOptions{})
+	defer wtb.Close()
+
+	config, err := wtbParser.ParseConfig([]byte(`
+{
+  "targets": {
+    "cluster_1": {
+      "weight":1,
+      "childPolicy": [{"round_robin": ""}]
+    }
+  }
+}`))
+	if err != nil {
+		t.Fatalf("failed to parse balancer config: %v", err)
+	}
+
+	// Send the config, and an address with hierarchy path ["cluster_1"].
+	addr := resolver.Address{Addr: testBackendAddrStrs[0], Attributes: nil}
+	if err := wtb.UpdateClientConnState(balancer.ClientConnState{
+		ResolverState:  resolver.State{Addresses: []resolver.Address{hierarchy.Set(addr, []string{"cluster_1"})}},
+		BalancerConfig: config,
+	}); err != nil {
+		t.Fatalf("failed to update ClientConn state: %v", err)
+	}
+
+	sc := <-cc.NewSubConnCh
+	wtb.UpdateSubConnState(sc, balancer.SubConnState{ConnectivityState: connectivity.TransientFailure})
```

Comment:
Check that the overall state is `TF` here before making the transition from `TF` to `Connecting`.

Reply:
Is it technically included in the below assertion. 

But, at this point, when the subConn first transitions to `TF`, if we don't check that the sub-balancer state is `TF`, then we are not actually testing what we want to test here (which is that a transition from `TF` to `Connecting` is suppressed).

You could basically copy over the same assertion which is done at the end of the test to here.